contrib/cirrus/runner.sh

```diff
@@ -247,6 +247,9 @@ dotest() {
 }
 
 msg "************************************************************"
+# BEFORE DOING ANYTHING AT ALL: make sure we have the env. needed for our job
+load_cirrus_task_environment
+
```

Comment:
This will be called multiple times...see around line 231 and possibly (rootless inside-container testing someday?) around line 116.   Maybe make the call conditional based on the current value of `$UID` and `$PRIV_NAME`?

Reply:
I'm really not sure how to interpret your feedback. Line 231 is literally the exact reason for all of this. So, of course we want `load_cirrus_task_environment` to be called in that case. And my invocation here is deliberate, because `runner.sh` is the common entry point and this `load` takes place before the subtask selection. 

To expand on my reasoning here: I found a bug in the current implementation. The current implementation adds environment variables to `.bashrc`, and this is a NOP on Ubuntu (not to mention bad practice, because it's a hidden side effect). My PR fixes that bug, by preserving necessary environment in a safe known place, then loading it at precisely the place where it's needed.

In reference to your later "have `lib.sh` always load it" comment, again I deliberately chose not to do that: `lib.sh` gets called from a lot of scripts, not all of which may want this side effect (but I haven't examined each one). I want `load_cirrus_task_environment` to be a deliberate intentional action requested by a script, in particular, by `runner.sh` which is the common delegating entry point. I want a future maintainer to see that and understand immediately that `runner.sh` is guaranteed to have the Cirrus envariables it relies on.

Could I ask you to take a fresh look? It's clear that one of us is missing something. If it's still me, could I ask you to reexplain your concerns with maybe more context that I may not be understanding?

Reply:
> Could I ask you to take a fresh look? 

Of course, anytime...